diff --git a/neuralop/losses/data_losses.py b/neuralop/losses/data_losses.py
index a2c05e5..d0f76c1 100644
--- a/neuralop/losses/data_losses.py
+++ b/neuralop/losses/data_losses.py
@@ -10,7 +10,7 @@ from typing import List
 
 import torch
 
-from .finite_diff import central_diff_1d, central_diff_2d, central_diff_3d
+from .finite_diff import central_diff_1d, central_diff_2d, central_diff_3d, central_diff_4d
 
 #loss function with rel/abs Lp loss
 class LpLoss(object):
@@ -222,15 +222,16 @@ class H1Loss(object):
         whether to fix finite difference derivative
         computation on the z boundary, by default False
     """
-    def __init__(self, d=1, measure=1., reduction='sum', eps=1e-8, fix_x_bnd=False, fix_y_bnd=False, fix_z_bnd=False):
+    def __init__(self, d=1, measure=1., reduction='sum', eps=1e-8, fix_x_bnd=False, fix_y_bnd=False, fix_z_bnd=False, fix_t_bnd=False):
         super().__init__()
 
-        assert d > 0 and d < 4, "Currently only implemented for 1, 2, and 3-D."
+        assert d > 0 and d < 5, "Currently only implemented for 1, 2, 3, and 4-D."
 
         self.d = d
         self.fix_x_bnd = fix_x_bnd
         self.fix_y_bnd = fix_y_bnd
         self.fix_z_bnd = fix_z_bnd
+        self.fix_t_bnd = fix_t_bnd
         
         self.eps = eps
         
@@ -263,6 +264,7 @@ class H1Loss(object):
             quadrature weights
 
         """
+
         dict_x = {}
         dict_y = {}
 
@@ -289,7 +291,7 @@ class H1Loss(object):
             dict_y[1] = torch.flatten(y_x, start_dim=-2)
             dict_y[2] = torch.flatten(y_y, start_dim=-2)
         
-        else:
+        elif self.d == 3:
             dict_x[0] = torch.flatten(x, start_dim=-3)
             dict_y[0] = torch.flatten(y, start_dim=-3)
 
@@ -303,6 +305,27 @@ class H1Loss(object):
             dict_y[1] = torch.flatten(y_x, start_dim=-3)
             dict_y[2] = torch.flatten(y_y, start_dim=-3)
             dict_y[3] = torch.flatten(y_z, start_dim=-3)
+
+        elif self.d == 4:
+            dict_x[0] = torch.flatten(x, start_dim=-4)
+            dict_y[0] = torch.flatten(y, start_dim=-4)
+
+            x_x, x_y, x_z, x_t = central_diff_4d(x, quadrature, fix_x_bnd=self.fix_x_bnd, fix_y_bnd=self.fix_y_bnd, fix_z_bnd=self.fix_z_bnd, fix_t_bnd=self.fix_t_bnd)
+            y_x, y_y, y_z, y_t = central_diff_4d(y, quadrature, fix_x_bnd=self.fix_x_bnd, fix_y_bnd=self.fix_y_bnd, fix_z_bnd=self.fix_z_bnd, fix_t_bnd=self.fix_t_bnd)
+
+
+            dict_x[1] = torch.flatten(x_x, start_dim=-4)
+            dict_x[2] = torch.flatten(x_y, start_dim=-4)
+            dict_x[3] = torch.flatten(x_z, start_dim=-4)
+            dict_x[4] = torch.flatten(x_t, start_dim=-4)
+
+            dict_y[1] = torch.flatten(y_x, start_dim=-4)
+            dict_y[2] = torch.flatten(y_y, start_dim=-4)
+            dict_y[3] = torch.flatten(y_z, start_dim=-4)
+            dict_y[4] = torch.flatten(y_t, start_dim=-4)
+
+        else:
+            raise NotImplementedError("H1Loss currently only implemented for 1, 2, 3 and 4-D.")
         
         return dict_x, dict_y
 
@@ -533,8 +556,8 @@ class HdivLoss(object):
 
             div_x = torch.flatten(x_x + x_y, start_dim=-2)
             div_y = torch.flatten(y_x + y_y, start_dim=-2)
-        
-        else:
+
+        elif self.d == 3:
             dict_x[0] = torch.flatten(x, start_dim=-3)
             dict_y[0] = torch.flatten(y, start_dim=-3)
 
@@ -543,6 +566,19 @@ class HdivLoss(object):
 
             div_x = torch.flatten(x_x + x_y + x_z, start_dim=-3)
             div_y = torch.flatten(y_x + y_y + y_z, start_dim=-3)
+
+        elif self.d == 4:
+            dict_x[0] = torch.flatten(x, start_dim=-4)
+            dict_y[0] = torch.flatten(y, start_dim=-4)
+
+            x_x, x_y, x_z, x_t = central_diff_4d(x, quadrature, fix_x_bnd=self.fix_x_bnd, fix_y_bnd=self.fix_y_bnd, fix_z_bnd=self.fix_z_bnd, fix_t_bnd=self.fix_t_bnd)
+            y_x, y_y, y_z, y_t = central_diff_4d(y, quadrature, fix_x_bnd=self.fix_x_bnd, fix_y_bnd=self.fix_y_bnd, fix_z_bnd=self.fix_z_bnd, fix_t_bnd=self.fix_t_bnd)
+
+            div_x = torch.flatten(x_x + x_y + x_z + x_t, start_dim=-4)
+            div_y = torch.flatten(y_x + y_y + y_z + y_t, start_dim=-4)
+
+        else:
+            raise NotImplementedError("HdivLoss currently only implemented for 1, 2, 3, and 4-D.")
         
         dict_x[1] = div_x
         dict_y[1] = div_y
@@ -761,4 +797,4 @@ class MSELoss(object):
         assert y_pred.shape == y.shape, (y.shape, y_pred.shape)
         if dim is None:
             dim = list(range(1, y_pred.ndim)) # no reduction across batch dim
-        return torch.mean((y_pred - y) ** 2, dim=dim).sum() # sum of MSEs for each element
\ No newline at end of file
+        return torch.mean((y_pred - y) ** 2, dim=dim).sum() # sum of MSEs for each element
